### PR TITLE
add early stopping

### PR DESCRIPTION
**Pull request by nthain**

### CODE REVIEW

File: src/model_tool.py [hunk 1]
```diff
@@ -130,10 +138,16 @@ def train(self, training_data_path, validation_data_path, text_column, label_col
...
+            earlyStopping = [EarlyStopping(min_delta=ES_MIN_DELTA, 
+                monitor='val_loss', patience=ES_PATIENCE, verbose=0, mode='auto')]
+        else:
+            earlyStopping = None
         print(self.model.fit(train_data, train_labels,
           batch_size=BATCH_SIZE,
           epochs=EPOCHS,
-          validation_data=(valid_data, valid_labels)))
+          validation_data=(valid_data, valid_labels),
+          callbacks = earlyStopping))
```
Comment (jetpack) [inline src/model_tool.py]: Nit: no spaces around the "="?

File: src/model_tool.py [hunk 2]
```diff
@@ -27,6 +30,9 @@
 EMBEDDING_DIM = 100
 EMBEDDING_TRAINABLE = False
 LEARNING_RATE = 0.00005
+STOP_EARLY = True
+ES_PATIENCE = 0 # Only relevant if STOP_EARLY = True
+ES_MIN_DELTA = 0 # Only relevant if STOP_EARLY = True
```
Comment (jetpack) [inline src/model_tool.py]: Are these reasonable defaults?

Comment (nthain) [reply src/model_tool.py]: Yes, I think they are until we've done further experimentation.

File: src/model_tool.py (hunk 1 above)
Comment (nthain) [reply src/model_tool.py]: Done - removed the spaces.

**Pull request merged**